Cleanup the introduction for the booleans concept (#1711)
exercism/go: concepts/booleans/introduction.md

# Introduction

> Booleans in Go are represented by the `bool` type, which values can be either `true` or `false`.

Reviewer: ```suggestion
Booleans in Go are represented by the `bool` type.
A `bool` is either `true` or `false`.
```

Split the sentence since there are two distinct statements rather than joining them with a comma.

Go supports three boolean operators: `!` (NOT), `&&` (AND), and `||` (OR).

```go
true || false // Output: true
true && false // Output: false
!true // Output: false
```

> The three boolean operators each have a different _operator precedence_. As a consequence, they are evaluated in this order: `!` first, `&&` second, and finally `||`. If you want to 'escape' these rules, you can enclose a boolean expression in parentheses (`()`), as the parentheses have an even higher operator precedence.

Reviewer: ```suggestion
The three boolean operators each have a different _operator precedence_.
As a consequence, they are evaluated in this order: `!` first, `&&` second, and finally `||`.
If you want to force a different ordering, you can enclose a boolean expression in parentheses (ie. `()`), as the parentheses have an even higher operator precedence.
```

Use one sentence per line and a couple other trivial changes.
